src/linker/Linker/UnconditionalSuppressMessageAttributeState.cs

```diff
@@ -41,17 +39,26 @@ public void AddLocalSuppression (CustomAttribute ca, ICustomAttributeProvider pr
 
 		public bool IsSuppressed (int id, MessageOrigin warningOrigin, out SuppressMessageInfo info)
 		{
-			if (HasLocalSuppressions && warningOrigin.MemberDefinition != null) {
-				IMemberDefinition memberDefinition = warningOrigin.MemberDefinition;
-				while (memberDefinition != null) {
-					if (IsLocallySuppressed (id, memberDefinition, out info))
-						return true;
+			info = default;
+			if (warningOrigin.MemberDefinition == null)
+				return false;
 
-					memberDefinition = memberDefinition.DeclaringType;
-				}
+			IMemberDefinition memberDefinition = warningOrigin.MemberDefinition;
+			while (memberDefinition != null) {
```

Comment:
Nit: I would use a separate local variable for the loop here - currently it's confusing.